### PR TITLE
A couple SW fixes

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: packages/service-worker/src/module.ts [hunk 1]
```diff
@@ -31,6 +31,14 @@ export function ngswAppInitializer(
     const isStable = op_take.call(onStable, 1) as Observable<boolean>;
     const whenStable = op_toPromise.call(isStable) as Promise<boolean>;
 
+    // Wait for service worker controller changes, and fire an ACTIVATE action when a new SW becomes
```
Comment (gkalpak) [inline packages/service-worker/src/module.ts]: Actually, it is an `INITIALIZE` action :grin:

**Review by gkalpak — COMMENTED** [review]
Is this still necessary:

https://github.com/angular/angular/blob/b922743f6e023a4fd76bf9b8e571e727073a1793/packages/service-worker/worker/src/driver.ts#L119-L131

If not, then this can probably be updated as well:

https://github.com/angular/angular/blob/b922743f6e023a4fd76bf9b8e571e727073a1793/packages/service-worker/worker/src/driver.ts#L195-L198

**Review by IgorMinar — COMMENTED** [review]
In https://github.com/angular/angular/pull/20715/commits/b8e60e287b1aaa2cf44d1d0f49d09080c625f82a you said that "The Service Worker" had a mechanism to keep itself alive, but it's not clear if you are referring to the platform api or to our implementation on top of it.

In any case, in that description you make it sound like you are replacing one way of keeping the service worker alive with this other way. But I don't see any code being removed, only code being added. Is this right?

File: packages/service-worker/src/module.ts (hunk 1 above)
Comment (IgorMinar) [reply packages/service-worker/src/module.ts]: @alxhub  can you please correct this?

**Review by IgorMinar — CHANGES_REQUESTED** [review]

**Review by IgorMinar — APPROVED** [review]
once corrected, please mark for merge.

**Comment by gkalpak** [comment]
I am still curious about https://github.com/angular/angular/pull/20715#pullrequestreview-80362233.

**Review by IgorMinar — APPROVED** [review]

File: packages/service-worker/src/module.ts (hunk 1 above)
Comment (alxhub) [reply packages/service-worker/src/module.ts]: Done.

**Comment by alxhub** [comment]
@gkalpak yes, we still want the SW to have that behavior.

**Pull request merged**